--- a/cmake/target_default_compile_warnings.cmake
+++ b/cmake/target_default_compile_warnings.cmake
@@ -10,10 +10,7 @@ function(target_default_compile_warnings_c THIS)
                 $<$<CONFIG:Debug>:-Wshadow>
                 $<$<CONFIG:Debug>:-Wunused>
                 -Wall -Wextra
-                -Wnon-virtual-dtor
-                -Wold-style-cast
                 -Wcast-align
-                -Woverloaded-virtual
                 -Wpedantic
                 -Wconversion
                 -Wsign-conversion
@@ -27,10 +24,7 @@ function(target_default_compile_warnings_c THIS)
                 $<$<CONFIG:Debug>:-Wshadow>
                 $<$<CONFIG:Debug>:-Wunused>
                 -Wall -Wextra
-                -Wnon-virtual-dtor
-                -Wold-style-cast
                 -Wcast-align
-                -Woverloaded-virtual
                 -Wpedantic
                 -Wconversion
                 -Wsign-conversion
